packages/amplify-graphql-conversation-transformer/src/__tests__/amplify-graphql-conversation-transformer.test.ts

```diff
@@ -44,6 +45,35 @@ describe('ConversationTransformer', () => {
       const schema = parse(out.schema);
       validateModelSchema(schema);
     });
+
+    it('uses functionMap for custom handler', () => {
+      const routeName = 'pirateChat';
+      const inputSchema = getSchema('conversation-route-custom-handler.graphql', { ROUTE_NAME: routeName });
+
+      const transformerManager = new TransformManager();
+      const stack = transformerManager.getTransformScope();
+      const customHandler = new Function(stack, 'conversation-handler', {
+        runtime: Runtime.NODEJS_18_X,
+        code: Code.fromInline('exports.handler = async (event) => { return "Hello World"; }'),
+        handler: 'index.handler',
+      });
+
+      const functionMap = {
+        [`Fn${routeName}`]: customHandler,
+      };
+
+      const out = transform(inputSchema, {}, defaultAuthConfig, functionMap, transformerManager);
+      expect(out).toBeDefined();
+
+      const expectedCustomHandlerArn = out.rawRootStack.resolve(customHandler.functionArn);
+      const conversationLambdaStackName = `${toUpper(routeName)}ConversationDirectiveLambdaStack`;
+      const conversationLambdaDataSourceName = `Fn${routeName}LambdaDataSource`;
+      const conversationLambdaDataSourceFunctionArnRef =
+        out.stacks[conversationLambdaStackName].Resources?.[conversationLambdaDataSourceName].Properties.LambdaConfig.LambdaFunctionArn.Ref;
+      const lambdaDataSourceFunctionArn =
+        out.rootStack.Resources?.[conversationLambdaStackName].Properties?.Parameters?.[conversationLambdaDataSourceFunctionArnRef];
+      expect(lambdaDataSourceFunctionArn).toEqual(expectedCustomHandlerArn);
```

Comment:
I welcome suggestions for a more succinct way to do this, like using CDK test helpers -- I was unable to find a better way.